spec/factories/case/ico_fois.rb

```diff
@@ -131,32 +131,33 @@
   factory :approved_ico_foi_case, parent: :pending_dacu_clearance_ico_foi_case do
     transient do
       identifier { 'approved ICO FOI case' }
+      date_draft_compliant { received_date + 2.days }
     end
 
-    date_draft_compliant { received_date + 2.days }
-
     after(:create) do |kase, evaluator|
       create :case_transition_approve,
              case: kase,
              acting_team: evaluator.approving_team,
              acting_user: evaluator.approver
 
       kase.approver_assignments.each { |a| a.update approved: true }
+      kase.update!(date_draft_compliant: evaluator.date_draft_compliant)
       kase.reload
     end
   end
 
   factory :responded_ico_foi_case, parent: :approved_ico_foi_case do
```

Comment:
Unsure of the reason for the transient + after_create loop - maybe a comment would help?